### PR TITLE
Feature integrate evm v0.35

### PR DESCRIPTION
**Pull request by gBaGu**
Contains evm v0.35 integration.

Depends on:
- https://github.com/velas/triedb/pull/8
- https://github.com/velas/evm/pull/3
- https://github.com/velas/parity-common/pull/2

Need to change dependencies repos after above PRs are merged.

### CODE REVIEW

File: evm-utils/evm-state/Cargo.toml [hunk 1]
```diff
@@ -5,17 +5,18 @@ authors = ["Vladimir Motylenko <vld@stegos.com>"]
...
-evm = { git = "https://github.com/velas/evm", branch = "add-traces-precompile-fix", features = ["with-serde"] }
+# TODO: change to velas/evm:master
+evm = { git = "https://github.com/gBaGu/evm", branch = "merge-upstream-master", features = ["with-serde"] }
 secp256k1 = { version = "0.19.0", features = ["recovery", "global-context"] }
 # force rand version, because 0.6.5 break compatibility with secp256k1
 rand2 = { version = "=0.6.1", package = "rand" }
 rocksdb = { git = "https://github.com/velas/rust-rocksdb", branch = "transaction", version = "0.17.0", default-features = false }
 # rocksdb = { version = "0.16.0", default-features = false }
-triedb = { git = "https://github.com/velas/triedb", branch = "feat/gc-simple", features = ["rocksdb"] }
+triedb = { git = "https://github.com/gBaGu/triedb", branch = "bump-primitive-types", features = ["rocksdb"] }
```
Comment (vldm) [inline evm-utils/evm-state/Cargo.toml]: Please create a tag in velas/triedb, all remaining prs was merged.

File: evm-utils/evm-state/src/context.rs [hunk 2]
```diff
@@ -284,6 +284,10 @@ where
         self.config.gas_limit.into()
     }
 
+    fn block_base_fee_per_gas(&self) -> U256 {
+        U256::zero()
```
Comment (vldm) [inline evm-utils/evm-state/src/context.rs:288]: Maybe we should set it to BURN_FEE?

File: rpc/src/evm_rpc_impl/mod.rs [hunk 3]
```diff
@@ -1168,10 +1182,9 @@ fn call_inner(
             Some(tx_chain_id),
             tx_hash,
             true,
-            solana_evm_loader_program::precompiles::simulation_entrypoint(
```
Comment (vldm) [inline rpc/src/evm_rpc_impl/mod.rs:1171]: Is there any reason to remove simulate_entrypoint?
I think we can avoid native accounts details in `call_` methods

File: evm-utils/evm-state/Cargo.toml (hunk 1 above)
Comment (gBaGu) [reply evm-utils/evm-state/Cargo.toml]: Done

File: rpc/src/evm_rpc_impl/mod.rs (hunk 3 above)
Comment (gBaGu) [reply rpc/src/evm_rpc_impl/mod.rs:1171]: Done

**Review by vldm — APPROVED** [review]

**Pull request merged**